sklearn/metrics/cluster/unsupervised.py

```diff
@@ -129,6 +141,11 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
         allowed by :func:`sklearn.metrics.pairwise.pairwise_distances`. If X is
         the distance array itself, use "precomputed" as the metric.
 
+    sample_weight : array, shape = [n_samples], optional
```

Comment:
I guess your test makes sure this is the right way to do it. It it not entirely obvious to me, though.